File: samples/CslaNet/cs/DeepData/DeepData.DAL.DLinq/OrderData.cs
using System.Data;
using System.Linq;
using Csla.Data;

namespace DeepData.DAL.DLinq
{
	///<summary>
	///</summary>
	public class OrderData : DeepData.DAL.OrderData
	{
		private DataClasses1DataContext db = new DataClasses1DataContext();

		///<summary>
		///</summary>
		///<returns></returns>
		public override object GetOrders()
		{
			var q = from o in db.Orders select o;
			IDbCommand command = db.GetCommand(q);
			command.Connection = db.Orders.Context.Connection;
			command.Connection.Open();
			SafeDataReader reader = (SafeDataReader)command.ExecuteReader(CommandBehavior.CloseConnection);
			return reader;
		}

		#region " IDisposable "

		///<summary>
		///</summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				db.Dispose();
			}
		}

		#endregion

	}
}

File: samples/CslaNet/vb/DeepData/DeepData.DAL.DLinq/OrderData.vb
Public Class OrderData
  Inherits DeepData.DAL.OrderData

  Private db As New DataClasses1DataContext

  Public Overrides Function GetOrders() As Object

    Dim q = From o In db.Orders
    Dim r() = q.ToArray
    Return r

  End Function

#Region " IDisposable "

  Protected Overrides Sub Dispose(ByVal disposing As Boolean)
    If disposing Then
      db.Dispose()
    End If
  End Sub

#End Region

End Class
